internal/cli/down.go

```diff
@@ -80,25 +82,55 @@ func (c *downCmd) down(ctx context.Context, downDeps DownDeps, args []string) er
 		return err
 	}
 
-	var dcProject model.DockerComposeProject
+	var dcServices []model.DockerComposeUpSpec
 	for _, m := range tlr.Manifests {
 		if m.IsDC() {
-			dcProject = m.DockerComposeTarget().Spec.Project
-			break
+			dcServices = append(dcServices, m.DockerComposeTarget().Spec)
 		}
 	}
 
-	if !model.IsEmptyDockerComposeProject(dcProject) {
+	if len(dcServices) > 0 {
 		dcc := downDeps.dcClient
-		err = dcc.Down(ctx, dcProject, logger.Get(ctx).Writer(logger.InfoLvl), logger.Get(ctx).Writer(logger.InfoLvl))
+		// at the time of writing, a Tiltfile can only have one DC project
+		proj := dcServices[0].Project
+
+		// `docker-compose down` removes all services in the config
+		// if we are only stopping a subset, we use `docker-compose rm` instead
+		// (don't *always* use `rm` because it doesn't remove networks)
```

Comment:
hmmmm...what's are the pros/cons of removing the networks? is it possible to garbage collect the networks ourselves?

i always get worried when there are two codepaths that are supposed to do the same thing but are implemented differently. Because then end-users on the team get very confused about why things are mysteriously failing, because it's non-obvious there are different codepaths.

but i guess it's fine for now, bc this unblocks things

Reply:
> what's are the pros/cons of removing the networks?

I don't have much confidence in an answer to this, but I suspect it doesn't matter much in practice. I mostly figured that behavior was worth preserving because people like to have a tool fully clean up after itself.

> is it possible to garbage collect the networks ourselves?

It looks like this is *mostly* `docker network prune --filter=label=com.docker.compose.project=$PROJECT_NAME`, but I haven't found a good way to get `$PROJECT_NAME`.

> but i guess it's fine for now, bc this unblocks things

I actually don't need the down.go changes to unblock anything, I just hoped it'd be a simple drive-by fix as long as I was adding `rm` support to the docker compose client. This is looking like a bigger yak than I want to shave at the moment, especially for an issue that no user has actually reported.

Reply:
I agree it's not probably worth shaving that yak right now. Just thought I'd point out that the [default project name is a normalized form of the name of the project directory](https://github.com/compose-spec/compose-go/blob/master/cli/options.go#L341-L350). I'm codifying this as well to make it easier to provide our own values for the project name in my PR #5259.